snyk-import/main.ts

```diff
@@ -0,0 +1,100 @@
+// deno-lint-ignore-file no-explicit-any
+
+import { ApiVersion, PhylumApi } from "phylum";
+import { parseArgs } from "https://deno.land/std@0.211.0/cli/parse_args.ts";
+import {
+  getOrgs,
+  getProjectDependencies,
+  getProjects,
+  Project,
+} from "./snyk-api.ts";
+import { setToken, snykToken } from "./token.ts";
+
+function usage() {
+  console.log(
+    "phylum snyk-import [--token SNYK_TOKEN] [--no-save-token] [--group phylum_group]",
+  );
+}
+
+const args = parseArgs(Deno.args, {
+  alias: { token: ["t"], group: ["g"] },
+  string: ["token", "group"],
+  boolean: ["save-token"],
+  negatable: ["save-token"],
+  default: { "save-token": true },
+  unknown: (arg) => {
+    console.error(`Unknown argument: ${arg}`);
+    usage();
+    Deno.exit(1);
+  },
+});
+
+let tokenArg = args.token;
+if (tokenArg && args["save-token"]) {
+  setToken(tokenArg);
+}
+
+if (!tokenArg) {
+  tokenArg = await snykToken();
+}
+
+const token = tokenArg;
+
+const orgs = await getOrgs(token);
+const projects = (await Promise.all(orgs.map((org) => getProjects(token, org))))
+  .flat();
+
+for (const project of projects) {
+  importProject(project).catch((err) => {
+    console.warn(
+      `Failed to import project '${project.name}': ${err}`,
+    );
+  });
+}
+
+async function importProject(project: Project) {
+  const dependencies = await getProjectDependencies(token, project);
+  if (dependencies.length === 0) {
+    console.warn(
+      `Skipping project '${project.name}'. No dependencies found.`,
+    );
+    return;
+  }
+
+  const createRes = await PhylumApi.createProject(project.name, args.group);
+  const projectId = createRes.id as string;
+  if (createRes.status == "Created") {
+    console.log(`Created phylum project '${project.name}'`);
+  }
+
+  const job_id = await PhylumApi.fetch(ApiVersion.V0, "/data/jobs/", {
+    method: "POST",
+    body: JSON.stringify({
+      packages: dependencies,
+      project: projectId,
+      label: "snyk-import",
+    }),
+  }).then(checkApiResponse).then(async (res: any) => {
+    const resp = await res.json();
+    return resp.job_id;
+  });
```

Comment:
What's the reason behind not using the extension API here? Are you trying to avoid converting the project name to an ID again?

Reply:
The extension API does not currently support submitting dependencies as PURLs, which is what I get from the SBOM

Reply:
I've opened phylum-dev/cli#1336 to track. Because the extension API probably should support this.